cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
I tried a similar test while waiting for an AMI build, and I think we'll need to remove `-e` from the shell options:

```
$ bash -c "trap 'echo OUCH' PIPE; while true; do sleep 10; echo here; done" &
[1] 31715
$ kill -SIGPIPE %1
OUCH
here
here
```

But with `-e`:
```
$ bash -e -c "trap 'echo OUCH' PIPE; while true; do sleep 10; echo here; done" &
[1] 31733
$ kill -SIGPIPE %1
OUCH
[1]+  Exit 141                bash -e -c "trap 'echo OUCH' PIPE; while true; do sleep 10; echo here; done"
```